src/Math-Core/PMVector.class.st

```diff
@@ -122,6 +122,18 @@ PMVector >> asPMVector [
 	^ self
 ]
 
+{ #category : #'as yet unclassified' }
+PMVector >> checkDimensionalCompatibility: dimensionArray [
+	|prod|
+	prod := 1.
+	
+	dimensionArray do: [ :each | prod := prod * each ].
+	
+	self assert: (self size = prod) description: 'Imcompatible combination of Dimensions provided'.
+	
+	^true 
```

Comment:
I think this is unnecessary as you're not relying on the output in places where you send this message.